MSAPI-79 | Fix Static Analyzer Warnings

AndriiYudaichev opened the pull request.

AndriiYudaichev committed:
MSAPI-79 | Fix Static Analyzer Warnings (Part 1)

CA1303 - Replace literal usage to resource file
CA1825 - Update way of zero length array initialization for .net standard2.0
CA1822 - Suppress - method logically instance dependant
CA1824 - Add NeutralResourcesLanguage attribute
CA1591 - Suppress XML documentation warnings on project level

AndriiYudaichev committed:
MSAPI-79 | Fix Static Analyzer Warnings (Part 2)

CA2237 - Suppress, warning applicable in case of few AppDomains
CA2227 - Suppress, warning not applicable, trying to follow it breakes deserialization
CA2000 - Refactor for code style consistency
CA1305 - Suppress, not applicable for ToString method

AndriiYudaichev committed:
MSAPI-79 | Fix Static Analyzer Warnings (Part 3)

CS1998 - Suppress till method not implemented
CA1305 - Suppress, n/a for unit tests
CA1303 - Suppress, n/a for unit tests
CA1823 - Delete unused fields

IvanBil commented on SignNow.Net/Exceptions/SignNowException.cs:
```diff
@@ -1,10 +1,12 @@
 
 namespace SignNow.Net.Exceptions
 {
+    [SuppressMessage("Microsoft.Usage", "CA2237:Mark with SerializableAttribute as Exception implements ISeriazable", Justification = "It is required in case of several AppDomains are used")]
```
А почему не сделать Serializable?

IvanBil commented on SignNow.Net/_Internal/Service/SignNowClient.cs:
```diff
@@ -74,14 +75,18 @@ private async Task<TResponse> RequestAsync<TResponse>(RequestOptions requestOpti
         /// <param name="response"></param>
         /// <exception cref="SignNowException">SignNow Exception.</exception>
         /// <returns></returns>
+        [SuppressMessage("", "CA1822:Method does not access instance data and can be marked as static", Justification = "Method is private and used inside instance method, it will not be used without the instance")]
```
почему не сделать статическим? В любом случае этот метод должен быть в другом классе, но пока что оставим его тут

IvanBil commented on SignNow.Net/_Internal/Service/SignNowClient.cs:
```diff
@@ -74,14 +75,18 @@ private async Task<TResponse> RequestAsync<TResponse>(RequestOptions requestOpti
                 var context = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
                 var apiError = context;
+#if NETSTANDARD2_0
+                var snException = Array.Empty<SignNowException>(); 
```
зачем это?

IvanBil commented on SignNow.Net/_Internal/Service/SignNowClient.cs:
```diff
@@ -113,11 +118,12 @@ private async Task ProcessErrorResponse(HttpResponseMessage response)
         /// <param name="requestOptions"></param>
         /// <exception cref="ArgumentException">The <paramref name="requestOptions">RequestUrl</paramref> argument is a null.</exception>
         /// <returns>Request Message <see cref="System.Net.Http.HttpRequestMessage"/></returns>
+        [SuppressMessage("", "CA1822:Method does not access instance data and can be marked as static", Justification = "Method is private and used inside instance method, it will not be used without the instance")]
```
Почему не сделать статическим?

IvanBil commented on SignNow.Net/OAuth2Service.cs:
```diff
@@ -80,11 +81,13 @@ public async Task<Token> GetTokenAsync(string code, Scope scope, CancellationTok
 
-        public Task<Token> RefreshTokenAsync(Token token, CancellationToken cancellationToken = default)
+        [SuppressMessage("Temporary", "CS1998:Method lacks await", Justification = "Suppressed till method not implemented")]
+        public async Task<Token> RefreshTokenAsync(Token token, CancellationToken cancellationToken = default)
```
предлагаю закомментить нереализованные методы

IvanBil commented on SignNow.Net/SignNow.Net.csproj:
```diff
@@ -19,6 +19,11 @@ SignNow allows you to embed legally-binding e-signatures into your app, CRM or c
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|netstandard1.2|AnyCPU'">
+      <NoWarn>1701;1702;1591</NoWarn>
```
это что?

AndriiYudaichev replied:
1701, 1702 - убрал, их VS по умолчанию подбрасывает, у нас таких ошибок нет
1591 - срабатывает если у публичного метода нет документации (при включенной генерации xml документации), нужно оставить

AndriiYudaichev committed:
MSAPI-79 | Fix Static Analyzer Warnings (Part 4)
 - Make fileds readonly
 - Rewrite MultipartFormDataContent initialization
 - Update assignment to compound assignment
 - Add throw to avoid JsonSerializationException suppressing
 - Update lambda to local function

AndriiYudaichev committed:
MSAPI-79 | Code Review Fixes
 - Add Serializable attribute for .net45 & .netstandard2.0
 - Globally suppress public members comments warning
 - Make methods static
 - Suppress CA1825 zero length array allocation warning
 - Comment RefreshTokenAsync, ValidateTokenAsync - will be implemented soon, comment preffarable to avoid exceptions on client side

AndriiYudaichev committed:
MSAPI-79 | Merge conflicts fix

AndriiYudaichev committed:
MSAPI-79 | Fix Warnings After Merge with Develop
 - add fileName and httpStatusCode xml documentation

IvanBil commented on SignNow.Net/_Internal/Service/SignNowClient.cs:
```diff
@@ -96,6 +97,7 @@ private async Task ProcessErrorResponse(HttpResponseMessage response)
                 }
                 catch (JsonSerializationException)
                 {
+                    throw;
```
тут не должно выбрасываться эксепшна

IvanBil committed:
[MSAPI-79] Fixed issue with unit tests. Now if SN returns 1 exception it is not passed as array.
